test/org/sagebionetworks/bridge/services/ScheduledActivityServiceMockTest.java

```diff
@@ -372,111 +380,154 @@ public void deleteActivitiesForSchedulePlanRejectsBadValue() {
     }
     
     @Test
     public void newActivitiesIncludedInSaveAndResultsV4() {
-        List<ScheduledActivity> scheduled = createNewActivities("AAA", "BBB");
-        List<ScheduledActivity> db = createStartedActivities("BBB");
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
         
-        List<ScheduledActivity> saves = merge(scheduled,db, V4_MERGE);
-        assertActivities(saves, "AAA");
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb));
         
-        scheduled = service.orderActivities(scheduled, V4_FILTER);
-        assertActivities(scheduled, "AAA", "BBB");
-    }
-    
-    private void assertActivities(List<ScheduledActivity> activities, String... guids) {
-        Set<String> results = Sets.newHashSet(guids);
-        assertEquals(results, toGuids(activities));
+        ScheduledActivityList list = new ScheduledActivityList(createStartedActivities("BBB"+TIME_PORTION), null, 0);
+        when(activityDao.getActivityHistoryV2(HEALTH_CODE, "BBB", NOW, NOW, TIME_ZONE, null, API_MAXIMUM_PAGE_SIZE))
+                .thenReturn(list);        
+        
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivitiesV4(createScheduleContext(NOW).build());
+        assertActivityGuids(returnedActivities, "AAA", "BBB");
+        
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "AAA");
     }
     
     @Test
     public void persistedAndScheduledIncludedInResultsV3() {
-        List<ScheduledActivity> scheduled = createNewActivities("CCC");
-        List<ScheduledActivity> db = createStartedActivities("CCC");
-
-        List<ScheduledActivity> saves = merge(scheduled, db, V3_MERGE);
-        assertActivities(saves);
+        SchedulePlan aaa = schedulePlan("AAA");
+        SchedulePlan bbb = schedulePlan("BBB");
+        SchedulePlan ccc = schedulePlan("CCC");
+        
+        when(schedulePlanService.getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY)).thenReturn(Lists.newArrayList(aaa,bbb,ccc));
+        when(activityDao.getActivities(eq(TIME_ZONE), any())).thenReturn(createStartedActivities("CCC"+TIME_PORTION));
+        
+        List<ScheduledActivity> returnedActivities = service.getScheduledActivities(createScheduleContext(NOW).build());
+        assertNotNull(returnedActivities.get(0).getStartedOn());
+        assertActivityGuids(returnedActivities, "AAA", "BBB", "CCC");
         
-        scheduled = service.orderActivities(scheduled, V3_FILTER);
-        assertNotNull(scheduled.get(0).getStartedOn());
+        verify(activityDao).saveActivities(scheduledActivityListCaptor.capture());
+        List<ScheduledActivity> saves = scheduledActivityListCaptor.getValue();
+        assertActivityGuids(saves, "AAA", "BBB");
```

Comment:
I'm about 95% sure the purpose of this test was to test what happens if we have no new activities to save. We should remove AAA and BBB from getSchedulePlans() to make sure this test tests the same thing as the original test.